### DISCUSSION TITLE
SteamDeck 如何安装wiliwili

### DISCUSSION THREAD

**Discussion by xfangfang** [xfangfang/wiliwili#41] [Q&A]
目前可以下载flatpak格式的安装包来安装，或直接从应用商店内搜索下载。

## 安装方法1 
直接应用商店搜索下载

## 安装方法2

如果应用商店没有搜索到，也可以尝试使用命令行下载：

命令行运行: `flatpak install flathub cn.xfangfang.wiliwili`

## 安装方法3
下载预编译的 flatpak 安装包

>  ⚠️ SteamDeck 是 x86 架构的处理器，不要下载成 ARM 架构的安装包

下载 wiliwili-Linux-**x86_64**.flatpak: https://github.com/xfangfang/wiliwili/releases

### 安装预编译的安装包
一般来说下载后双击安装包即可安装。

也可以使用命令行的安装方式：
```shell
sudo flatpak install wiliwili-Linux-x86_64.flatpak
```
默认是安装在系统目录，需要用户密码才能安装，你也可以安装在用户目录
```shell
flatpak install --user wiliwili-Linux-x86_64.flatpak
```
---

注1：如果你之前设置过系统密码，那么安装的过程中会询问系统密码，这是正常的现象。

注2：如果你使用应用商店安装过其他应用，那么大概率会立刻安装成功，如果这是你首次安装flatpak应用，那么可能需要下载约200M大小的基础环境。  

注3：如果是首次安装flatpak应用，在系统应用列表中可能不会立刻出现图标，重启（或者重新登录）后会正常显示在系统菜单中，在这之前你也可以通过命令行的方式，或者从应用商店的已安装列表中打开 wiliwili。

## 如何运行
安装好之后，系统中影音分区就会出现wiliwili，点击打开即可。

如果你想命令行运行，请执行：
```shell
flatpak run cn.xfangfang.wiliwili
```

## 如何卸载
在商店的已安装列表中选中 wiliwili 点击 Remove 即可。如果你需要删除用户数据，在点击卸载后从同一个页面就能找到删除用户数据的按钮。
> 用户数据位于：$HOME/.var/app/cn.xfangfang.wiliwili/config/wiliwili

如果你想命令行运行，请执行：
```shell
flatpak uninstall cn.xfangfang.wiliwili
```

## 如何更新
下载新的安装包，双击打开，在打开的应用商店页面点击右上角 Remove，等到卸载后再点击 Install 即可。放心操作，你的用户数据并不会因此被删除。

如果你愿意，也可以使用命令行的方式直接安装更新，方法和首次安装是一样的。

## 如何执行命令
前面提到的命令行运行，可以在系统应用中查找一个叫 `Konsole` 的软件，打开后向里面输入内容即可，回车表示输入结束

## 扩展阅读
[PC客户端自定义字体或按键图标](https://github.com/xfangfang/wiliwili/discussions/38)  
[WIKI](https://github.com/xfangfang/wiliwili/wiki)

**Comment by xfangfang** [comment]
~补充：已知问题，在首页不要按y搜索，会卡住，因为搜索页暂时还没有做适配 2022/12/06~

更新：最新版支持了pc端的系统输入法 2023/01/09

**Comment by OuNiSang** [comment]
![WiliWili_Icon](https://user-images.githubusercontent.com/30402947/236590987-4e8b5e14-6baa-4759-81b2-62ab098472d6.png)
![bilibili_capsule](https://user-images.githubusercontent.com/30402947/236590988-f3fa0fdb-802a-4767-b01d-e7b3b4c04286.png)
![bilibili_Icon](https://user-images.githubusercontent.com/30402947/236590989-33247ce9-f024-4ca3-9cbf-3ba942ad3173.png)
![bilibili_banner](https://user-images.githubusercontent.com/30402947/236590990-3b7846d9-574a-4727-ae2f-4a453e26bc74.png)

Uploading some image that could be helpful for setting up in steamdeck

**Comment by lakejason0** [comment]
由于Steam封面需要的有横板和竖版，正方形的裁剪似乎有些略丑，我便稍微用GIMP扩展了一下（把渐变补上了）：
![Wiliwili steam cover](https://github.com/xfangfang/wiliwili/assets/36039861/20c53015-bfcd-4bb4-b8b7-e293a77c8d26)
![Wiliwili steam cover vertical](https://github.com/xfangfang/wiliwili/assets/36039861/5a3e723e-743a-4674-bd37-56652e6ca3ac)

**Comment by honjow** [comment]
上flathub是要给 https://github.com/flathub/flathub 提pr吧？
有这方面的相关进展吗

**Reply by xfangfang** [reply]
据我所知还没有人做这个事

**Reply by honjow** [reply]
昨天看了下要求。因为不太清楚flatpak的构建，有点摸不着头脑🤣

**Reply by pugaizai** [reply]
https://github.com/xfangfang/wiliwili/blob/yoga/.flatpak-manifest.yml 这个就是flatpak的构建文件

**Reply by xfangfang** [reply]
@honjow 在 [madoka773](https://github.com/madoka773) 的帮助下，已经上架了 flathub：`flatpak install flathub cn.xfangfang.wiliwili`

**Comment by honjow** [comment]
> @honjow 在 [madoka773](https://github.com/madoka773) 的帮助下，已经上架了 flathub：`flatpak install flathub cn.xfangfang.wiliwili`
好好好
